code/client/makecatalogs

```diff
@@ -150,6 +202,27 @@ def makecatalogs(repopath, options):
                 if key.startswith('_'):
                     del(pkginfo[key])
 
+            if iconhashing:
+
+                name = pkginfo.get('name')
+
+                if pkginfo.get('icon_name'):
+                    if icons.get(pkginfo.get('icon_name')):
+                        iconhash = icons.get(pkginfo.get('icon_name'))
+                    elif icons.get(pkginfo.get('icon_name')+'.png'):
+                        iconhash = icons.get(pkginfo.get('icon_name')+'.png')
+                    else:
+                        errors.append("WARNING: icon_name specified"
+                        " in info file %s but it does not exist"
+                        % filepath[len(pkgsinfopath)+1:])
+                else:
+                    if icons.get(name+'.png'):
+                        iconhash = icons.get(name+'.png')
```

Comment:
Same here; this should be just:

```
iconhash = icons.get(name + '.png')
```

without the if. icon hash will get set to a value or None. No need to call `icons.get(name+'.png')` twice.